spring-cloud-gcp-autoconfigure/src/main/java/com/google/cloud/spring/autoconfigure/sql/CloudSqlEnvironmentPostProcessor.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright 2017-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.spring.autoconfigure.sql;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.google.cloud.sql.CredentialFactory;
+import com.google.cloud.sql.core.CoreSocketFactory;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.env.EnvironmentPostProcessor;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.MapPropertySource;
+import org.springframework.util.ClassUtils;
+
+/**
+ * Provides Google Cloud SQL instance connectivity through Spring JDBC by providing only a
+ * database and instance connection name.
+ *
+ * @author João André Martins
+ * @author Artem Bilan
+ * @author Mike Eltsufin
+ * @author Chengyuan Zhao
+ * @author Eddú Meléndez
+ */
+public class CloudSqlEnvironmentPostProcessor implements EnvironmentPostProcessor {
+	private final static String CLOUD_SQL_PROPERTIES_PREFIX = "spring.cloud.gcp.sql.";
+
+	private static final Log LOGGER =
+			LogFactory.getLog(CloudSqlEnvironmentPostProcessor.class);
+
+	@Override
+	public void postProcessEnvironment(ConfigurableEnvironment environment, SpringApplication application) {
+		DatabaseType databaseType = getEnabledDatabaseType(environment);
+
+		if (databaseType != null) {
+			if (LOGGER.isInfoEnabled()) {
+				LOGGER.info("post-processing Cloud SQL properties for + " + databaseType.name());
+			}
+
+			CloudSqlJdbcInfoProvider cloudSqlJdbcInfoProvider = buildCloudSqlJdbcInfoProvider(environment, databaseType);
+
+			// configure default JDBC driver and username as fallback values when not specified
+			Map<String, Object> fallbackMap = new HashMap<>();
+			fallbackMap.put("spring.datasource.username", "root");
```

Comment:
Should we be giving a default at all? It's `root` for MySQL, but [not for Postgres](https://cloud.google.com/sql/docs/postgres/users#default-users).

Reply:
I think we should set the default as to promote the notion of convention over configuration.
But, yeah good catch on Postgres having a different username.
It's sort of breaking change, but we can do it for 2.0.